django_socio_grpc/tests/fakeapp/handlers.py

```diff
@@ -28,8 +29,8 @@
 urlpatterns = []
 
 
-def grpc_handlers(server):
-    app_registry = AppHandlerRegistry("fakeapp", server)
+def grpc_handlers(server, directory: str = None):
+    app_registry = AppHandlerRegistry("fakeapp", server, proto_directory=directory)
```

Comment:
why this changes ?

Reply:
we need the directory to get the proto in the AppHandlerRegistry for the option only_generate_non_annotated_for_new_fields()

Reply:
if we run generateproto with a different directory we need this parameter to get the good one

Reply:
I don't understand why it was not needed before and now it is. I even don't understand the concept behind: only_generate_non_annotated_for_new_fields

How both are related ? Why non annotated annotated new field need a directory ?

Reply:
but i think before reviewing we should make sure we are on the same page for the implementation, i think Léni still has doubts concerning the idea of non annotated generation in itself

Reply:
Without this it makes all choices introduce big breaking changes

Reply:
I really don't see how it is creating breaking change. I think I miss an inplem details. 

The comparaison of existing value should be in the generators not in the plugin.

It would be better to talk directly in french on a call. But i am not in the slack anymore. Maybe you can create a call meet ?

Reply:
well, if we update dsg on an app backend, all choices defined as "string" in the .proto will be converted to an Enum if they are buildable. This is an automatic breaking change.
That's why we'd like to have a way to only automatically generate Enum for new choice fields, and for this we need to access the previous data in the plugin so it can decide whether to generate a ProtoEnum or not.
I'm not against doing a call

Reply:
Okay I understand now the whole thing. Indeed the fact to transform enum should not be creating breaking change before version 1.
We just have to remove the enum plugin from the defautl settings.

Reply:
Yes so maybe we could just add a parameter in the plugin to completly deactivate generation from non annotated, this will also prevent breaking change when updating to 1.0.
And to be clear, you are against the idea of using a parameter only_generate_non_annotated_for_new_fields ?

Reply:
Yes, The choice should be using the plugin or not using it not using it only for new fields.
If user want to do it only for some action he can add the generation plugin only for the action he want

Reply:
Okay, and you would be okay with the idea of a boolean to enable/disable non annotated generation ? So that a user that has a preference for declaritive way versus an automatic way, that could or could not generate the Enum if the choices values are not well formatted, can still use the plugin ?

Reply:
@AMontagu

Reply:
Okay for me. Need a comment and a section in the doc to say that it will be defaulting in 1.0.0.